### ISSUE TITLE
Compilation fails with gfortran 10

### ISSUE THREAD

**Issue by wgnoehring** [libAtoms/QUIP#209]
Compiling quippy with gfortran 10.1.1 and `QUIP_ARCH=linux_x86_64_gfortran` fails with the following error message:
```
<path_to_QUIP>/QUIP/src/Utils/force_machine_learning_module.f95:691:39:

  688 |        call fread_array_d(1, distance_matrix(1,1), trim(data_dir)//'Dist.dat')
      |                             2          
......
  691 |               call  fread_array_d(i-1, distance_matrix(i,:(i-1)), trim(data_dir)//'Dist.dat')
      |                                       1
Error: Rank mismatch between actual argument at (1) and actual argument at (2) (scalar and rank-1)
<path_to_QUIP>/QUIP/src/Utils/force_machine_learning_module.f95:643:27:

  643 |     call fwrite_array_d(1, 0.0_dp, trim(data_dir)//'Dist.dat')
      |                           1
......
  675 |    call fwrite_array_d(size(distance_confs_unsorted), distance_confs_unsorted(:), trim(data_dir)//'Dist.dat')
      |                                                      2
Error: Rank mismatch between actual argument at (1) and actual argument at (2) (rank-1 and scalar)
```
The error can be downgraded to a warning by adding `-fallow-argument-mismatch`  to the Fortran compiler flags in Makefile.rules, but using this flag is discouraged.

**Comment by jameskermode** [comment]
Thanks for reporting. This module is actually deprecated and can be removed from the compilation without loss of important functionality, so I'll do this.

**Comment by jameskermode** [comment]
Should be fixed by 4e6ef645991
